src/parser/parser.yy

```diff
@@ -1532,10 +1532,6 @@ match_edge_type_list
         $$ = $1;
         $$->add($3);
     }
-    | match_edge_type_list PIPE COLON name_label {
```

Comment:
It's legal syntax.

Reply:
```
neo4j@neo4j> match (v:teacher)-[e:like|:serve]-(n) return n;
The semantics of using colon in the separation of alternative relationship types in conjunction with
the use of variable binding, inlined property predicates, or variable length is no longer supported.
Please separate the relationships types using `:A|B|C` instead (line 2, column 18 (offset: 18))
"match (v:teacher)-[e:like|:serve]-(n) return n;"
```

Reply:
As I known, we don't follow the newest version of open cypher. @CPWstatic Which version we followed?